### ISSUE TITLE
output:floats default value

### ISSUE THREAD

**Issue by johnomotani** [boutproject/BOUT-dev#34]
In lines 257-8 of bout++.cxx the default value of floats is set to true

```
if(!options->getSection("output")->isSet("floats"))
  options->getSection("output")->set("floats", true, "default"); // by default output floats
```

This contradicts (and overrides) the default value in Datafile, which is confusing (I would not expect to find the default values for output options set in BoutInitialise).

I propose these lines be deleted.

(If the default then becomes false, then examples/test-io could probably have the tolerance reduced from 1.e-10, since the I/O would then use double precision? Possibly the benchmark data should be re-generated as well at higher precision)

**Comment by bendudson** [comment]
This is a good point, and this probably shouldn't be here. The rationale was that restart files should be double precision by default, but it's unusual that outputs need to be in double precision. This was a crude attempt to do the "right thing" by default.

On balance I agree that this isn't how the default should be overridden, and these lines should be deleted.

Suggest just adding the setting

[output]
floats = true

to examples/test-io to avoid having to regenerate benchmark data.

**Comment by johnomotani** [comment]
It is true that even if you want to output floats to the dump files, you still want doubles in the restart files. Maybe it would be best to have separate Options sections for 'output' and 'restart'? Then 'downgrading' the output would not affect the restart files as a side-effect.

**Comment by bendudson** [comment]
The restart files do use a different set of options, in a section "restart"
rather than "output". Just setting output:floats=true should only degrade
the dmp files, not the restart files.

**Comment by johnomotani** [comment]
Ah yes, I should have remembered that.

**Comment by ZedThree** [comment]
Fixed in `v4.0.0-RC`
